Products/CMFPlone/browser/icon.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+from OFS.Image import File
+from plone.registry.interfaces import IRegistry
+from Products.CMFCore.interfaces import ISiteRoot
+from zope.component import adapter
+from zope.component import getUtility
+from zope.component.hooks import getSite
+from zope.interface import implementer
+from zope.interface import Interface
+from zope.location.interfaces import LocationError
+from zope.traversing.interfaces import ITraversable
+from zope.publisher.interfaces import IPublishTraverse
+
+
+@implementer(IPublishTraverse)
+class IconView:
+
+    prefix = 'plone.staticresources.icon.'
+
+    def publishTraverse(self, request, name):
+        request['TraversalRequestNameStack'] = []
+        return self
+
+    def __init__(self, context, request):
+        super(IconView, self).__init__(context, request)
+        if len(self.request.path) == 1:
+            self.icon = request.path[0]
+
+    def __call__(self, icon=None):
```

Comment:
API wise I would put the code in here in a `def url(....` and use it in `__call__` (if at all). But when used like an API it read better:
```
<div tal:define="icon_view nocall:context/@@icon">
   <img src="${python:icon_view.url('alarm')" />
   <p><tal:icon tal:replace="structure python:icon_view.tag('archive', tag_alt='foobar')" /></p>
```